### ISSUE TITLE
Graph color and position appears to be off.

### ISSUE THREAD

**Issue by jdmaruschak** [abosmet/pywxfades#1]
Three different graphics, three different appearances. We need to figure out which combination is an accurate plot and figure out why the program isn't getting there (or is getting there).
Solving this issue will require manual analysis of the raw data for this model run and comparing these data to the internal data tracking.

Image 1: Legacy python code running on Inferno. The graphic appears to be proper and complete. 
Graph colors do not line up with either the new code or the GrADS code. Referencing the lowest precipitation value line at 18Z, the line is red until 15Z, then green until 21Z, then blue.
The graph appears to be showing an extra time-step at the beginning when compared to the GrADS code. The GrADS code seems to start a 12Z which is likely the cause of this discrepancy.
![allentown pa](https://cloud.githubusercontent.com/assets/7535452/5763716/c9f99c4e-9cba-11e4-885a-df9af3f380f0.png)

Image 2: Current code running on Inferno. The colors at the beginning of the graph appear broken.
The lowest precipitation value line is colored purple until 12Z, then red until 15Z, then green until 21Z, then blue.
Graph positioning is similar to the GrADS code, but the precipitation timing appears to be shifted an hour into the past. The pillar of dots at 00Z on this graph is consistent with the pillar of dots at 03Z on the GrADS chart.
![precip-type-allentown pa](https://cloud.githubusercontent.com/assets/7535452/5763798/6ee39264-9cbb-11e4-9fb5-14952c4ff224.png)

Image 3: Rich's GrADS code running on eyewall.met Graphic appears proper and complete.
I do not like the graph starting at 12Z, however. I also want to make sure this is working as intended.
![](http://eyewall.met.psu.edu/rich/gefs/20150112/srefgifs/09Z/Allentown,PA-ptype.png)

Basically this boils down to, we need to push the graph to the right by 1 time-step.

**Comment by jdmaruschak** [comment]
sref_nmm.t09z.pgrb132.p3.grib2 is the model file which holds the data for the line referenced above. We can use this file to validate the plotting method as it is easily identifiable.

**Comment by jdmaruschak** [comment]
![test](https://cloud.githubusercontent.com/assets/7535452/5770471/1bb64a6c-9cf9-11e4-8e0e-54bb8e9fd3a8.png)

I created the above output manually. The spikey bar at the top is an error and it is tedious to redo. Suffice to say I was able to plot the graph correctly manually and now we need to translate it to procedural code.

**Comment by jdmaruschak** [comment]
![precip-type-allentown pa](https://cloud.githubusercontent.com/assets/7535452/5771036/49aef18a-9d03-11e4-8d9e-351df0a64916.png)

Updated image 2 including the sref-em bug fix.

**Comment by jdmaruschak** [comment]
Manual data exploration output:

./sref_nmm.t09z.pgrb132.p3.grib2 Using forecastTime
print fcst_index,msg.forecastTime,msg.validityTime,msg.shortName,value,accum,msg.startStep,msg.endStep
print fcst_index,msg.forecastTime,msg.validityTime,msg.shortName,value,msg.startStep,msg.endStep
0 0 900 tp 0.0 0.0 0 0
0 0 1200 tp 0.0 0.0 0 3
1 3 1500 tp 0.5 0.01968505 3 6
2 6 1500 cfrzr 1.0 6 6
2 6 1800 tp 1.625 0.0836614625 6 9
3 9 1800 crain 1.0 9 9
3 9 2100 tp 1.75 0.1525591375 9 12
4 12 2100 crain 1.0 12 12
4 12 0 tp 2.5 0.2509843875 12 15
5 15 0 csnow 1.0 15 15
5 15 300 tp 2.375 0.344488375 15 18
6 18 300 csnow 1.0 18 18
6 18 600 tp 0.125 0.3494096375 18 21
7 21 900 tp 0.0 0.3494096375 21 24
8 24 1200 tp 0.0 0.3494096375 24 27
9 27 1500 tp 0.0 0.3494096375 27 30
10 30 1800 tp 0.0 0.3494096375 30 33
11 33 2100 tp 0.0 0.3494096375 33 36
12 36 0 tp 0.0 0.3494096375 36 39
13 39 300 tp 0.0 0.3494096375 39 42
14 42 600 tp 0.0 0.3494096375 42 45
15 45 900 tp 0.0 0.3494096375 45 48
16 48 1200 tp 0.0 0.3494096375 48 51
17 51 1500 tp 0.0 0.3494096375 51 54
18 54 1800 tp 0.0 0.3494096375 54 57
19 57 2100 tp 0.0 0.3494096375 57 60
20 60 0 tp 0.0 0.3494096375 60 63
21 63 300 tp 0.0 0.3494096375 63 66
22 66 600 tp 0.0 0.3494096375 66 69
23 69 900 tp 0.0 0.3494096375 69 72
24 72 1200 tp 0.0 0.3494096375 72 75
25 75 1500 tp 0.0 0.3494096375 75 78
26 78 1800 tp 0.0 0.3494096375 78 81
27 81 2100 tp 0.0 0.3494096375 81 84
28 84 0 tp 0.0 0.3494096375 84 87
PYTHON STOP

./sref_nmm.t09z.pgrb132.p3.grib2 Using endStep
print fcst_index,msg.forecastTime,msg.validityTime,msg.shortName,value,accum,msg.startStep,msg.endStep
print fcst_index,msg.forecastTime,msg.validityTime,msg.shortName,value,msg.startStep,msg.endStep
0 0 900 tp 0.0 0.0 0 0
1 0 1200 tp 0.0 0.0 0 3
2 3 1500 tp 0.5 0.01968505 3 6
2 6 1500 cfrzr 1.0 6 6
3 6 1800 tp 1.625 0.0836614625 6 9
3 9 1800 crain 1.0 9 9
4 9 2100 tp 1.75 0.1525591375 9 12
4 12 2100 crain 1.0 12 12
5 12 0 tp 2.5 0.2509843875 12 15
5 15 0 csnow 1.0 15 15
6 15 300 tp 2.375 0.344488375 15 18
6 18 300 csnow 1.0 18 18
7 18 600 tp 0.125 0.3494096375 18 21
8 21 900 tp 0.0 0.3494096375 21 24
9 24 1200 tp 0.0 0.3494096375 24 27
10 27 1500 tp 0.0 0.3494096375 27 30
11 30 1800 tp 0.0 0.3494096375 30 33
12 33 2100 tp 0.0 0.3494096375 33 36
13 36 0 tp 0.0 0.3494096375 36 39
14 39 300 tp 0.0 0.3494096375 39 42
15 42 600 tp 0.0 0.3494096375 42 45
16 45 900 tp 0.0 0.3494096375 45 48
17 48 1200 tp 0.0 0.3494096375 48 51
18 51 1500 tp 0.0 0.3494096375 51 54
19 54 1800 tp 0.0 0.3494096375 54 57
20 57 2100 tp 0.0 0.3494096375 57 60
21 60 0 tp 0.0 0.3494096375 60 63
22 63 300 tp 0.0 0.3494096375 63 66
23 66 600 tp 0.0 0.3494096375 66 69
24 69 900 tp 0.0 0.3494096375 69 72
25 72 1200 tp 0.0 0.3494096375 72 75
26 75 1500 tp 0.0 0.3494096375 75 78
27 78 1800 tp 0.0 0.3494096375 78 81
28 81 2100 tp 0.0 0.3494096375 81 84
29 84 0 tp 0.0 0.3494096375 84 87
PYTHON STOP

**Comment by jdmaruschak** [comment]
Using endStep rather than forecastTime appears to produce the effect we want.

**Comment by jdmaruschak** [comment]
Graphic produced using endStep instead of forecastTime. This issue is closed.

![precip-type-allentown pa](https://cloud.githubusercontent.com/assets/7535452/5780056/811db76a-9d75-11e4-9536-4ce120ab1026.png)
